Helper/ScopeSetting.php

```diff
@@ -0,0 +1,195 @@
+            $this->_state->setAreaCode(\Magento\Framework\App\Area::AREA_ADMINHTML);
+        }
+    }
+
+    /**
+     * Getter method for a given scope setting
+     * @param string $path
+     */
+    protected function getScopeSetting($path)
+    {
+        $this->checkAreaCode();
+
+        if ($this->_state->getAreaCode() == \Magento\Framework\App\Area::AREA_ADMINHTML) {
+            $scopedStoreCode = $this->_request->getParam('store');
+            $scopedWebsiteCode = $this->_request->getParam('website');
+        } else {
+            // In frontend area. Only concerned with store for frontend.
+            $scopedStoreCode = $this->_storeId;
+        }
+
+        if (isset($scopedStoreCode)) {
+            $scope = \Magento\Store\Model\ScopeInterface::SCOPE_STORE;
+            return $this->_scopeConfig->getValue($path, $scope, $scopedStoreCode);
+        } elseif (isset($scopedWebsiteCode)) {
+            $scope = \Magento\Store\Model\ScopeInterface::SCOPE_WEBSITE;
+            return $this->_scopeConfig->getValue($path, $scope, $scopedWebsiteCode);
+        } else {
+            return $this->_scopeConfig->getValue($path);
+        };
+    }
+
+    /**
+     * Setter method for a given scope setting
+     * @param string $path
+     * @param mixed $value
+     */
+    protected function setScopeSetting($path, $value)
+    {
+        $this->checkAreaCode();
+
+        if ($this->_state->getAreaCode() == \Magento\Framework\App\Area::AREA_ADMINHTML) {
+            $scopedStoreCode = $this->_request->getParam('store');
+            $scopedWebsiteCode = $this->_request->getParam('website');
+        } else {
+            // In frontend area. Only concerned with store for frontend.
+            $scopedStoreCode = $this->_storeId;
+        }
+
+        if (isset($scopedStoreCode)) {
+            $scope = \Magento\Store\Model\ScopeInterface::SCOPE_STORE;
+            return $this->_configWriter->save($path, $value, $scope, $scopedStoreCode);
+        } elseif (isset($scopedWebsiteCode)) {
+            $scope = \Magento\Store\Model\ScopeInterface::SCOPE_WEBSITE;
+            return $this->_configWriter->save($path, $value, $scope, $scopedWebsiteCode);
+        } else {
+            return $this->_configWriter->save($path, $value);
+        };
+    }
+
+    public function getVersion()
+    {
+        return $this->_moduleList
+            ->getOne(self::MODULE_NAME)['setup_version'];
+    }
+
+    public function getEnabled()
+    {
+        return $this->getScopeSetting(self::ENABLE);
+    }
+
+    public function getPublicApiKey()
+    {
+        return $this->getScopeSetting(self::PUBLIC_API_KEY);
+    }
+
+    public function getPrivateApiKey()
+    {
+        return $this->getScopeSetting(self::PRIVATE_API_KEY);
+    }
+
+    public function setPrivateApiKey($value)
+    {
+        return $this->setScopeSetting(self::PRIVATE_API_KEY, $value);
+    }
+
+    public function getLoggerEnabled()
```

Comment:
maybe rename this function to isLoggerEnabled if this is going to return a boolean.